common/objectcachev2.h

```diff
@@ -136,27 +125,13 @@ class ObjectCacheV2 {
                 if (x->rc == 0) {
                     // make vector holds those shared_ptr
                     // prevent object destroy in critical zone
-#if __cplusplus < 201703L
                     to_release.push_back(x->reset());
```

Comment:
Like here, have to reset & get old value as a atomic instruction, so that it can be collected for destruction.